batcher/src/lib.rs

```diff
@@ -170,42 +183,75 @@ impl App {
         Ok(())
     }
 
-    pub async fn verify_sp1_proof(&self, proof: &[u8], elf: &[u8]) -> Result<(), anyhow::Error> {
-        let (_pk, vk) = self.sp1_prover_client.setup(elf);
-        let proof = bincode::deserialize(proof).map_err(|_| anyhow::anyhow!("Invalid proof"))?;
-
-        self.sp1_prover_client
-            .verify(&proof, &vk)
-            .map_err(|_| anyhow::anyhow!("Failed to verify proof"))?;
+    async fn add_task(self: &Arc<Self>, verification_data: VerificationData) {
+        info!("Adding verification data to batch...");
 
-        Ok(())
-    }
+        let len = {
+            let mut current_batch = self.current_batch.lock().await;
+            current_batch.push(verification_data);
 
-    pub async fn add_task(&self, verification_data: VerificationData) {
-        info!("Adding verification data to batch...");
+            debug!("Batch size: {}", current_batch.len());
 
-        let mut current_batch = self.current_batch.lock().await;
-        current_batch.push(verification_data);
+            current_batch.len()
+        };
 
-        debug!("Batch size: {}", current_batch.len());
-        if current_batch.len() < 2 {
-            return;
+        if len >= self.batch_size_interval {
+            let c = self.clone();
+            tokio::spawn(async move {
+                let block_number = c
+                    .eth_rpc_provider
+                    .get_block_number()
+                    .await
+                    .expect("Failed to get block number");
+                let block_number =
+                    u64::try_from(block_number).expect("Failed to convert block number");
+                c.handle_new_block(block_number).await;
+            });
         }
+    }
 
-        let batch_merkle_tree: MerkleTree<VerificationBatch> = MerkleTree::build(&current_batch);
-        let batch_merkle_root_hex = hex::encode(batch_merkle_tree.root);
-        info!("Batch merkle root: {}", batch_merkle_root_hex);
+    async fn handle_new_block(&self, block_number: u64) {
+        let (batch_bytes, batch_merkle_root) = {
+            let mut last_uploaded_batch_block = self.last_uploaded_batch_block.lock().await;
+            let mut current_batch = self.current_batch.lock().await;
+            let current_batch_len = current_batch.len();
+            if current_batch_len <= 1 {
+                // Needed because merkle tree freezes on only one leaf
+                info!("New Block reached but current batch is empty or has only one task");
```

Comment:
This shouldn't be happening

Reply:
We can fix it in another PR but we should make an issue

Reply:
created issue https://github.com/yetanotherco/aligned_layer/issues/190